### PR TITLE
GUI for adding links

### PR DESCRIPTION
**Pull request by stephendade**
This PR adds a GUI for adding links by modifying the link module's menu to include a custom dialog.

Tested in Ubuntu20 and Windows10.

Screenshot from 2021-01-31 15-20-19

### CODE REVIEW

File: MAVProxy/mavproxy.py [hunk 1]
```diff
@@ -1340,8 +1340,7 @@ def quit_handler(signum = None, frame = None):
           print("Connecting to %s" % serial_list[0])
           mpstate.module('link').link_add(serial_list[0].device)
     elif not opts.master and len(serial_list) > 1:
-          print("Error: multiple possible serial ports; use --master to select a single port")
-          sys.exit(1)
+          print("Warning: multiple possible serial ports. Use console GUI to add port")
```
Comment (peterbarker) [inline MAVProxy/mavproxy.py:1343]: The old advice here is still good.
```suggestion
          print("Warning: multiple possible serial ports. Use console GUI or 'link add'  to add port, or restart using --master to select a single port")
```

File: MAVProxy/modules/lib/wx_addlink.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    '*FTDI*',
+    "*Arduino_Mega_2560*",
+    "*3D*",
+    "*USB_to_UART*",
+    '*Ardu*',
+    '*PX4*',
+    '*Hex_*',
+    '*Holybro_*',
+    '*mRo*',
+    '*FMU*']
```
Comment (peterbarker) [inline MAVProxy/modules/lib/wx_addlink.py:22]: This is going to get out-of-date with the other list.  Perhaps we drop this into a state structure or something?

File: MAVProxy/modules/mavproxy_link.py [hunk 3]
```diff
@@ -208,6 +207,11 @@ def link_add(self, descriptor, force_connected=False):
         '''add new link'''
         try:
             (device, optional_attributes) = self.parse_link_descriptor(descriptor)
+            # if there's only 1 colon in device, it must be serialdevice:baud
+            # As all network connections will be pre-pended with tcp:, udpin:, etc
```
Comment (peterbarker) [inline MAVProxy/modules/mavproxy_link.py:211]: That's not the case; `localhost:1234` is the same as `udpin:localhost:1234`

**Review by peterbarker — CHANGES_REQUESTED** [review]
This tests out well for me.

The only interface niggle I found is that when creating a serial connection the default baudrate is 38,400 - useful to nobody :-)  (well, that and the "remove link" dropdown is completely busted, but that's not your problem here)

File: MAVProxy/mavproxy.py (hunk 1 above)
Comment (stephendade) [reply MAVProxy/mavproxy.py]: Fixed

File: MAVProxy/modules/lib/wx_addlink.py (hunk 2 above)
Comment (stephendade) [reply MAVProxy/modules/lib/wx_addlink.py]: Removed and now importing preferred_ports from ``MAVProxy/modules/mavproxy_link.py``

File: MAVProxy/modules/mavproxy_link.py (hunk 3 above)
Comment (stephendade) [reply MAVProxy/modules/mavproxy_link.py]: Ahh, did not did consider that. Have fixed up such to check for that case

**Comment by stephendade** [comment]
@peterbarker: I've updated the PR and it should be good to merge now.

**Comment by peterbarker** [comment]
Looks OK to me, but this is a significant behavioural change on startup so @tridge is going to have to be good with it.

**Review by peterbarker — APPROVED** [review]

**Comment by tridge** [comment]
if no DISPLAY env variable I think we should fail as before if no devices

**Review by tridge — CHANGES_REQUESTED** [review]
need to check DISPLAY on linux

**Comment by stephendade** [comment]
Have added check for DISPLAY

**Review by tridge — APPROVED** [review]

**Pull request merged**